### PR TITLE
feat: add Python implementation of accelerated OP

### LINKED ISSUE #36: [Feature Request] Decrease the installation time of TorchOpt on GPU machine
## Motivation

It may take 3-4min to install TorchOpt, seems unnecessary for such small package

## Solution

Maybe optimize the cmakefile.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### LINKED ISSUE #66: [Feature Request] Pre-built wheels for macOS and Windows
## Motivation

Please outline the motivation for the proposal.
Is your feature request related to a problem? e.g., "I'm always frustrated when [...]".
If this is related to another issue, please link here too.

macOS and Windows support.

## Solution

Update `CMakeList.txt` and `cibuildwheel`.

## Checklist

- [X] I have checked that there is no similar issue in the repo (**required**)

### PR DESCRIPTION
**Pull request by XuehaiPan**
## Description

Resolves #36
Resolves #66

## Motivation and Context

- [X] I have raised an issue to propose this change ([required](https://github.com/metaopt/torchopt/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [X] New feature (non-breaking change which adds core functionality)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [X] I have read the [CONTRIBUTION](https://torchopt.readthedocs.io/en/latest/developer/contributing.html) guide (**required**)
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [X] I have reformatted the code using `make format` (**required**)
- [X] I have checked the code using `make lint` (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

**Review by Benjamin-eecs — COMMENTED** [review]
LGTM!

File: torchopt/accelerated_op/_src/adam_op.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    count: int,
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    """Adam backward updates."""
+    one_minus_pow_b1 = 1.0 - pow(b1, count)
+    inv_one_minus_pow_b2 = 1.0 / (1.0 - pow(b2, count))
+
+    updates_div_new_mu = updates.div(new_mu)
+    denominator = updates_div_new_mu.mul(one_minus_pow_b1)
+    dnew_mu_out = dupdates.mul(updates_div_new_mu)
+    dnew_nu_out = dupdates.mul(updates).mul_(denominator.square()).mul_(-0.5 * inv_one_minus_pow_b2)
```
Comment (JieRen98) [inline torchopt/accelerated_op/_src/adam_op.py:109]: use '._square()'

File: torchopt/accelerated_op/_src/adam_op.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    b2: float,
+    count: int,
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    """Adam backward updates."""
+    one_minus_pow_b1 = 1.0 - pow(b1, count)
+    inv_one_minus_pow_b2 = 1.0 / (1.0 - pow(b2, count))
+
+    updates_div_new_mu = updates.div(new_mu)
+    denominator = updates_div_new_mu.mul(one_minus_pow_b1)
+    dnew_mu_out = dupdates.mul(updates_div_new_mu)
```
Comment (JieRen98) [inline torchopt/accelerated_op/_src/adam_op.py:108]: use updates_div_new_mu.mul_(updates)

File: torchopt/accelerated_op/_src/adam_op.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    return new_mu.mul(inv_one_minus_pow_b1).div_(
+        new_nu.mul(inv_one_minus_pow_b2).add_(eps_root).sqrt_().add_(eps)
+    )
+
+
+def backward_mu(
+    dmu: torch.Tensor, updates: torch.Tensor, mu: torch.Tensor, b1: float
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    """Adam backward mu."""
+    dupdates = dmu.mul(1.0 - b1)
```
[the remark this answers is not in the document]
Comment (XuehaiPan) [reply torchopt/accelerated_op/_src/adam_op.py:79]: It will be cast to the same floating point type as `dmu.dtype`, usually `torch.float32`.

**Review by Benjamin-eecs — DISMISSED** [review]
LGTM

**Pull request merged**